ccpp/data/GFS_typedefs.F90

```diff
@@ -2027,6 +2036,7 @@ subroutine statein_create (Statein, IM, Model)
     allocate (Statein%pgr    (IM))
     allocate (Statein%ugrs   (IM,Model%levs))
     allocate (Statein%vgrs   (IM,Model%levs))
+    allocate (Statein%wgrs   (IM,Model%levs))
```

Comment:
Is this wgrs only used when the Model%lightning_threat is .true.? If yes, may we only allocate it when  Model%lightning_threat is .true.?

Reply:
Yes. The atmos_cubed_sphere will have to know about that, though. It is updated by the dynamical core.

Reply:
My question is: Why not use existing state variable vvl (omega pa/s) to calcuate wgrs by vvl = dp/dz*wgrs?

Reply:
I would like the developer of the CCPP code, @welewis69, to answer that question.

Reply:
Chunxi and others, our only concern regarding using wgrs as solved for from vvl = dp/dz*wgrs would be that the lightning threat indices are based on empirical relationships derived using the non-hydrostatic vertical velocity. In FV3, is the fully non-hydrostatic w recoverable from vvl in this manner?

Reply:
In FV3, non-hydyostatic w is retrievable from vvl. The only issue is vvl is smoothed with the second-order smoothing. It is smoothed once by default in dynamics.

Reply:
Thank you for the clarification, Chunxi. I appreciate it.

Reply:
@SamuelTrahanNOAA @ChunxiZhang-NOAA @dustinswales So, where did we land on this issue? Are we going to keep it as-is, or simplify and be consistent with other physics?

Reply:
@welewis69 - Can you respond again?  Should we stay with the unsmoothed non-hydrostatic vertical velocity, or derive it from the smoothed omega?